data/src/test/java/org/apache/iceberg/data/TestLocalScan.java

```diff
@@ -175,67 +173,110 @@ private void overwriteExistingData() throws IOException {
+    File location = temp;
+    Assertions.assertThat(location.delete()).isTrue();
     this.sharedTableLocation = location.toString();
     this.sharedTable =
         TABLES.create(
             SCHEMA,
             PartitionSpec.unpartitioned(),
-            ImmutableMap.of(TableProperties.DEFAULT_FILE_FORMAT, format.name()),
+            ImmutableMap.of(TableProperties.DEFAULT_FILE_FORMAT, temp.getName()),
             sharedTableLocation);
 
     Record record = GenericRecord.create(SCHEMA);
 
-    DataFile file1 =
-        writeFile(sharedTableLocation, format.addExtension("file-1"), file1FirstSnapshotRecords);
-
-    Record nullData = record.copy();
-    nullData.setField("id", 11L);
-    nullData.setField("data", null);
-
-    DataFile file2 =
-        writeFile(sharedTableLocation, format.addExtension("file-2"), file2FirstSnapshotRecords);
-    DataFile file3 =
-        writeFile(sharedTableLocation, format.addExtension("file-3"), file3FirstSnapshotRecords);
-
-    // commit the test data
-    sharedTable.newAppend().appendFile(file1).appendFile(file2).appendFile(file3).commit();
+    String[] format = {"parquet", "orc", "avro"};
+    Arrays.stream(format)
+        .forEach(
+            str -> {
+              FileFormat fileFormat = FileFormat.fromString(str);
+              DataFile file1 = null;
+              try {
+                file1 =
+                    writeFile(
+                        sharedTableLocation,
+                        fileFormat.addExtension("file-1"),
+                        file1FirstSnapshotRecords);
+              } catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+              Record nullData = record.copy();
+              nullData.setField("id", 11L);
+              nullData.setField("data", null);
+
+              DataFile file2 = null;
+              try {
+                file2 =
+                    writeFile(
+                        sharedTableLocation,
+                        fileFormat.addExtension("file-2"),
+                        file2FirstSnapshotRecords);
+              } catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+              DataFile file3 = null;
+              try {
+                file3 =
+                    writeFile(
+                        sharedTableLocation,
+                        fileFormat.addExtension("file-3"),
+                        file3FirstSnapshotRecords);
+              } catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+
+              // commit the test data
+              sharedTable
+                  .newAppend()
+                  .appendFile(file1)
+                  .appendFile(file2)
+                  .appendFile(file3)
+                  .commit();
+            });
   }
 
-  @Test
-  public void testRandomData() throws IOException {
+  @ParameterizedTest(name = "format = {0}")
+  @MethodSource("data")
```

Comment:
this should be parameterized at the class level. You can check `TableMetadataParserTest` as an example for how to do that. Test methods would then get the `@TestTemplate` annotation